### PR TITLE
WIP: ci: add a possibility to run tests

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
This an attempt to add a basic per buildsystem run tests command. Hopefully it works

### CODE REVIEW

**Comment by tchx84** [comment]
> it's pretty fast to build, & would be just part of the devel version of the manifest?

Sounds reasonable!

Related the last failed build:

```
File "/usr/local/bin/rewrite-flatpak-manifest", line 41, in <module>
    build_options["env"]["DISPLAY"] = "0:0"
KeyError: 'env'
```
:thinking: 

Thanks a lot for looking into this, let me know how can I assist you.

**Comment by bilelmoussaoui** [comment]
> > it's pretty fast to build, & would be just part of the devel version of the manifest?
> 
> Sounds reasonable!
> 
> Related the last failed build:
> 
> ```
> File "/usr/local/bin/rewrite-flatpak-manifest", line 41, in <module>
>     build_options["env"]["DISPLAY"] = "0:0"
> KeyError: 'env'
> ```
> 
> thinking
> 
> Thanks a lot for looking into this, let me know how can I assist you.

That's just a broken thing on the rewrite-flatpak-manifest script I use. To add `run-tests: true` to the latest module defined on the manifest. Though, the tests kept failing as I have to start x11 somehow but as I only ask flatpak to run the tests, i'm not sure how can I achieve that ^^

**Comment by tchx84** [comment]
> > > it's pretty fast to build, & would be just part of the devel version of the manifest?
> > 
> > 
> > Sounds reasonable!
> > Related the last failed build:
> > ```
> > File "/usr/local/bin/rewrite-flatpak-manifest", line 41, in <module>
> >     build_options["env"]["DISPLAY"] = "0:0"
> > KeyError: 'env'
> > ```
> > 
> > 
> > thinking
> > Thanks a lot for looking into this, let me know how can I assist you.
> 
> That's just a broken thing on the rewrite-flatpak-manifest script I use. To add `run-tests: true` to the latest module defined on the manifest. Though, the tests kept failing as I have to start x11 somehow but as I only ask flatpak to run the tests, i'm not sure how can I achieve that ^^

I wonder what it is asking for that in the first place, I suppose some GTK backend... Maybe there's something to change in the tests to prevent this?

**Comment by bilelmoussaoui** [comment]
No idea, but I'm more in favour to figure out how to run a x11 server so that everyone can enjoy their tests without having to modify anything :)

**Comment by tchx84** [comment]
> No idea, but I'm more in favour to figure out how to run a x11 server so that everyone can enjoy their tests without having to modify anything :)

I support this message :smile:

**Comment by tchx84** [comment]
```
Unable to init server: Could not connect: Connection refused

(gjs:17): Gtk-WARNING **: 16:57:08.579: cannot open display: 
```

Progress!

**Comment by bilelmoussaoui** [comment]
Ok X server is up and running now :) the rest is on your side :P

**Comment by tchx84** [comment]
> Ok X server is up and running now :) the rest is on your side :P

haha, awesome!! :clap:

**Comment by tchx84** [comment]
EDIT:  nvm, forgot I could do it myself hah

**Comment by bilelmoussaoui** [comment]
Nice! I'm cleaning up the action part & then i will tag a release

**Comment by tchx84** [comment]
> Nice! I'm cleaning up the action part & then i will tag a release

:clap: 

btw, which branch should it use? I see it is currently commented out.

**Comment by bilelmoussaoui** [comment]
> > Nice! I'm cleaning up the action part & then i will tag a release
> 
> clap
> 
> btw, which branch should it use? I see it is currently commented out.

That part is to allow tests running when you push to a branch without opening a PR. I always set it to master only because otherwise tests are ran (for the push & pull requests events) twice for each PR.

```

1/4 Validate desktop file   OK             0.01s
2/4 Validate appstream file OK             0.48s
3/4 Validate schema file    OK             0.01s
4/4 Jasmine tests           OK             11.17s

Ok:                 4   
Expected Fail:      0   
Fail:               0   
Unexpected Pass:    0   
Skipped:            0   
Timeout:            0   
```
seems to be ready o/

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 on:
   push:
-    branches: [master]
+    # branches: [master]
```
Comment (tchx84) [inline .github/workflows/CI.yml]: I see, we should we leave it like this then ? :point_up:

Comment (bilelmoussaoui) [reply .github/workflows/CI.yml]: Nop, fixed it

**Comment by bilelmoussaoui** [comment]
This should be ready :)

**Comment by tchx84** [comment]
> This should be ready :)

Thanks @bilelmoussaoui this is great!

**Pull request merged**